File: ToDo.Backend/ToDo.Backend.API/Serivce/ToDoService.cs
using ToDo.Backend.API.Interface;
using ToDo.Backend.API.Models;

namespace ToDo.Backend.API.Serivce
{
    public class ToDoService : IToDoService
    {
        public Task AddAsync(ToDoItem item)
        {
            throw new NotImplementedException();
        }

        public Task<IEnumerable<ToDoItem>> GetAllAsync()
        {
            throw new NotImplementedException();
        }

        public Task<ToDoItem?> GetByIdAsync(int id)
        {
            throw new NotImplementedException();
        }

        public Task SaveChangesAsync()
        {
            throw new NotImplementedException();
        }

        public Task UpdateAsync(ToDoItem item)
        {
            throw new NotImplementedException();
        }
    }
}

File: ToDo.Frontend/ToDo.View/src/app/services/todo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, Subject  } from 'rxjs';
import { ToDoItem } from '../models/todo-item.model';

@Injectable({ providedIn: 'root' })
export class ToDoService {
  private apiUrl = 'http://localhost:5296/api/todo';

  tasksUpdated = new Subject<void>();

  constructor(private http: HttpClient) {}

  getAll(): Observable<ToDoItem[]> {
    return this.http.get<ToDoItem[]>(this.apiUrl);
  }

  add(item: Partial<ToDoItem>): Observable<ToDoItem> {
    return this.http.post<ToDoItem>(this.apiUrl, item);
  }

  getById(id: number): Observable<ToDoItem> {
    return this.http.get<ToDoItem>(`${this.apiUrl}/${id}`);
  }

    update(task: ToDoItem): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/${task.id}`, task);
  }
    notifyUpdate() {
    this.tasksUpdated.next();
  }

}
